components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2318,28 +2318,63 @@ SRC_PATH="$PLAN_CONTEXT"
 # Expand the path of this program to an absolute path
 THIS_PROGRAM=$(abspath "$0")
 
-# Now to ensure a `plan.sh` exists where we expect it. There are 2 possible
-# candidate locations relative to the `$PLAN_CONTEXT` directory: a `./plan.sh`
-# or a `./habitat/plan.sh`. Only one or the other location is allowed so that
-# a Plan author isn't confused if they edit one to have this program read
-# the other.
+# Now to ensure a `plan.sh` exists where we expect it. There are 4 possible
+# candidate locations relative to the `$PLAN_CONTEXT` directory: 
+#   `./plan.sh`
+#   `./habitat/plan.sh`
+#   `./$pkg_target/plan.sh
+#   `./habitat/$pkg_target/plan.sh 
+# In most cases, only one location is allowed so that a Plan author isn't 
+# confused if they edit one to have this program read the other.
+# The exception to this is when the $pkg_target requires variations to
+# the default `plan.sh`. Plan authors can create these variants by placeing 
+# Plan contents in a subdirectory named the same as the intended $pkg_target. 
+# In this case, we want to check if there is a variant `plan.sh` available 
+# for the $pkg_target, and prefer loading that one. Alternate $pkg_target 
+# are not considered as part of this load logic.
+
+# With plan variants, plans can exist in 4 places per $pkg_target relative to 
+# the `$PLAN_CONTEXT` directory. Only two combinations are allowed:
+#
+#   `./plan.sh` AND `./$pkg_target/plan.sh` 
+#   OR 
+#   `./habitat/plan.sh` AND `./habitat/$pkg_target/plan.sh`
+# Consider all other combination of two plans and abort if found.
```

Comment:
Should this say `Consider all other combinations of two plans invalid and abort if found` ?